src/hooks/usePositioning.tsx

```diff
@@ -1,74 +1,99 @@
-import {ElementRef, MutableRefObject, useEffect, useRef, useState} from 'react';
+import {MutableRefObject, useEffect, useRef, useState} from 'react';
 import toPX from 'to-px';
 
-type TPosition = {
+type Position = {
     top?: number;
     left?: number;
     bottom?: number;
     right?: number;
 }
-type TAnchorElOrigin = {
+type AnchorElOrigin = {
     horizontal: 'left' | 'center' | 'right';
     vertical: 'top' | 'center' | 'bottom';
 };
-type TTransformElOrigin = {
+type TransformElOrigin = {
     horizontal: 'left' | 'right';
     vertical: 'top' | 'bottom';
 };
 
-const initialPosition: TPosition = {
+const initialPosition: Position = {
     top: -1000,
     left: -1000
 };
 
-function getPosition(anchorPosition: TPosition): TPosition {
+function getPosition(anchorPosition: Position): Position {
     return {
         top: typeof anchorPosition.top === 'string' ? toPX(anchorPosition.top) : anchorPosition.top,
         left: typeof anchorPosition.left === 'string' ? toPX(anchorPosition.left) : anchorPosition.left
     };
 }
 
+function getClosestRelativeAncestor(el: HTMLElement) {
+    while (el.parentElement) {
+        if (window.getComputedStyle(el).position === 'relative') {
```

Comment:
I would rather use window.getComputedStyle(el).position !== 'static' , as absolute/fixed/sticky are also positioned

Reply:
Ah yes, good catch. I'll change this.